### DISCUSSION TITLE
v3.0.0-beta.4

### DISCUSSION THREAD

**Discussion by joschi** [dropwizard/dropwizard#6066] [Announcements]
## What's Changed
* Update upgrade notes for Dropwizard 2.1.x regarding the Grizzly connector by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5492
* Update dependency org.hibernate:hibernate-core to v5.6.10.Final by @renovate in https://github.com/dropwizard/dropwizard/pull/5504
* Update dependency org.apache.httpcomponents.core5:httpcore5 to v5.1.4 by @renovate in https://github.com/dropwizard/dropwizard/pull/5507
* Update dependency org.checkerframework:checker-qual to v3.23.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5514
* Update dependency org.liquibase:liquibase-core to v4.13.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5515
* Update actions/setup-java digest to 2c7a487 by @renovate in https://github.com/dropwizard/dropwizard/pull/5525
* Update actions/cache action to v3.0.5 by @renovate in https://github.com/dropwizard/dropwizard/pull/5526
* Update github/codeql-action digest to 3e7e3b3 by @renovate in https://github.com/dropwizard/dropwizard/pull/5531
* Update dependency org.jdbi:jdbi3-bom to v3.31.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5544
* Update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.4.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5542
* Update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.4.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5543
* Update dependency org.cyclonedx:cyclonedx-maven-plugin to v2.7.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5560
* Update dependency org.apache.maven.plugins:maven-deploy-plugin to v3 by @renovate in https://github.com/dropwizard/dropwizard/pull/5571
* Update dependency org.apache.maven.plugins:maven-deploy-plugin to v3 by @renovate in https://github.com/dropwizard/dropwizard/pull/5570
* Update dependency org.apache.maven.plugins:maven-install-plugin to v3 by @renovate in https://github.com/dropwizard/dropwizard/pull/5572
* Update dependency org.apache.maven.plugins:maven-install-plugin to v3 by @renovate in https://github.com/dropwizard/dropwizard/pull/5573
* Bump actions/stale from 5.0.0 to 5.1.0 by @dependabot in https://github.com/dropwizard/dropwizard/pull/5548
* Update dependency org.apache.tomcat:tomcat-jdbc to v10.0.23 by @renovate in https://github.com/dropwizard/dropwizard/pull/5602
* Update dependency mysql:mysql-connector-java to v8.0.30 by @renovate in https://github.com/dropwizard/dropwizard/pull/5594
* Update dependency Sphinx to v5.1.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5595
* Update dependency org.liquibase:liquibase-core to v4.14.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5584
* Update dependency org.apache.maven.plugins:maven-resources-plugin to v3.3.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5597
* Update dependency org.jdbi:jdbi3-bom to v3.32.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5583
* Update dependency org.apache.maven.plugins:maven-resources-plugin to v3.3.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5596
* Update dependency Sphinx to v5.1.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5613
* Update junit5 monorepo to v5.9.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/5614
* Fix tests by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5618
* Update github/codeql-action digest to 0c670bb by @renovate in https://github.com/dropwizard/dropwizard/pull/5628
* Update actions/stale digest to 9c1b1c6 by @renovate in https://github.com/dropwizard/dropwizard/pull/5627
* Update dependency net.bytebuddy:byte-buddy to v1.12.13 by @renovate in https://github.com/dropwizard/dropwizard/pull/5636
* Replace `org.glassfish.hk2.external:jakarta.inject` with `jakarta.inject:jakarta.inject-api` by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5638
* Update github/codeql-action digest to 2ca79b6 by @renovate in https://github.com/dropwizard/dropwizard/pull/5654
* Update dependency com.uber.nullaway:nullaway to v0.9.9 by @renovate in https://github.com/dropwizard/dropwizard/pull/5655
* Update dependency org.apache.maven.plugins:maven-site-plugin to v3.12.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5656
* Update dependency org.apache.maven.plugins:maven-site-plugin to v3.12.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5657
* Update dependency org.hibernate.validator:hibernate-validator to v6.2.4.Final by @renovate in https://github.com/dropwizard/dropwizard/pull/5658
* Update error_prone.version to v2.15.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/5660
* Update dependency org.checkerframework:checker-qual to v3.24.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5659
* Update dependency jp.skypencil.errorprone.slf4j:errorprone-slf4j to v0.1.13 by @renovate in https://github.com/dropwizard/dropwizard/pull/5670
* Update dependency io.dropwizard.metrics:metrics-bom to v4.2.11 by @renovate in https://github.com/dropwizard/dropwizard/pull/5674
* Update actions/cache action to v3.0.6 by @renovate in https://github.com/dropwizard/dropwizard/pull/5678
* Update dependency org.liquibase:liquibase-core to v4.15.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5685
* Properly handle `null` in PortRangeValidator by @joschi in https://github.com/dropwizard/dropwizard/pull/5690
* Add toJavaDuration to Duration by @sleberknight in https://github.com/dropwizard/dropwizard/pull/5692
* Update dependency org.javassist:javassist to v3.29.1-GA by @renovate in https://github.com/dropwizard/dropwizard/pull/5702
* Update actions/cache action to v3.0.7 by @renovate in https://github.com/dropwizard/dropwizard/pull/5701
* Update mockito monorepo to v4.7.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/5710
* Remove guava exclusion from mustache dependency by @rhowe in https://github.com/dropwizard/dropwizard/pull/5712
* Remove deprecated EnvironmentVariableLookup by @rhowe in https://github.com/dropwizard/dropwizard/pull/5713
* Update dependency org.apache.maven.plugins:maven-javadoc-plugin to v3.4.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5724
* Update dependency org.apache.maven.plugins:maven-javadoc-plugin to v3.4.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5723
* Wrap exceptions thrown by beforeStart with MappableException by @andref in https://github.com/dropwizard/dropwizard/pull/5506
* Update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.4.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5730
* Update dependency org.apache.maven.plugins:maven-help-plugin to v3.3.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5731
* Update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.4.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5729
* Update github/codeql-action digest to f5d217b by @renovate in https://github.com/dropwizard/dropwizard/pull/5740
* Allow parameter conversion exceptions to be mapped to responses by exception mappers by @tmelmoth in https://github.com/dropwizard/dropwizard/pull/5691
* Update github/codeql-action digest to 7fee4ca by @renovate in https://github.com/dropwizard/dropwizard/pull/5756
* Update actions/cache action to v3.0.8 by @renovate in https://github.com/dropwizard/dropwizard/pull/5757
* Update dependency net.bytebuddy:byte-buddy to v1.12.14 by @renovate in https://github.com/dropwizard/dropwizard/pull/5758
* Update github/codeql-action digest to c7f292e by @renovate in https://github.com/dropwizard/dropwizard/pull/5767
* Remove redundant outputPatternAsHeader logging setting from tests by @rhowe in https://github.com/dropwizard/dropwizard/pull/5770
* Suppress logging during dropwizard-validations tests by @rhowe in https://github.com/dropwizard/dropwizard/pull/5716
* Minimise test logging configuration by @rhowe in https://github.com/dropwizard/dropwizard/pull/5771
* Use Byte Buddy instead of Javassist  in UnitOfWorkAwareProxyFactory by @sjthomason in https://github.com/dropwizard/dropwizard/pull/5748
* Update logback to version 1.3.0 by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5781
* Update dependency com.uber.nullaway:nullaway to v0.9.10 by @renovate in https://github.com/dropwizard/dropwizard/pull/5793
* Update github/codeql-action digest to b398f52 by @renovate in https://github.com/dropwizard/dropwizard/pull/5792
* Update dependency jp.skypencil.errorprone.slf4j:errorprone-slf4j to v0.1.15 by @renovate in https://github.com/dropwizard/dropwizard/pull/5794
* Address errorprone warnings in dropwizard-logging by @rhowe in https://github.com/dropwizard/dropwizard/pull/5778
* Update dependency org.hibernate:hibernate-core to v5.6.11.Final by @renovate in https://github.com/dropwizard/dropwizard/pull/5795
* Update dependency org.checkerframework:checker-qual to v3.25.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5806
* Update dependency com.fasterxml.jackson:jackson-bom to v2.13.4 by @renovate in https://github.com/dropwizard/dropwizard/pull/5805
* Update dependency io.dropwizard.metrics:metrics-bom to v4.2.12 by @renovate in https://github.com/dropwizard/dropwizard/pull/5810
* Exclude `jetty-servlet-api` by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5814
* Re-introduce automatic module names by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5813
* Update dependency org.glassfish.jersey:jersey-bom to v2.37 by @renovate in https://github.com/dropwizard/dropwizard/pull/5819
* Update upgrade notes for 3.0.0 by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5822
* Update actions/setup-java digest to d854b6d by @renovate in https://github.com/dropwizard/dropwizard/pull/5831
* Update dependency net.bytebuddy:byte-buddy to v1.12.16 by @renovate in https://github.com/dropwizard/dropwizard/pull/5832
* Only configure JMX, if the JMXConfigurator is present by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5825
* Update mockito monorepo to v4.8.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/5833
* Allow parameter conversion exceptions to be mapped to responses by exception mappers by @tmelmoth in https://github.com/dropwizard/dropwizard/pull/5838
* Update dependency com.uber.nullaway:nullaway to v0.10.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5843
* Update dependency org.liquibase:liquibase-core to v4.16.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5844
* Disable check for static keyword on SLF4J loggers by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5725
* Drop support for Java 8 in dropwizard-jersey by @rhowe in https://github.com/dropwizard/dropwizard/pull/5855
* Update dependency com.uber.nullaway:nullaway to v0.10.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5860
* Update dependency org.hibernate.validator:hibernate-validator to v6.2.5.Final by @renovate in https://github.com/dropwizard/dropwizard/pull/5861
* Exchange HTTP client implementation by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5872
* Improve startup performance for many loggers by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5854
* Update link to Jackson's polymorphic deserialization docs by @bskarda in https://github.com/dropwizard/dropwizard/pull/5882
* Update dependency org.eclipse.jetty:jetty-bom to v10.0.12 by @renovate in https://github.com/dropwizard/dropwizard/pull/5899
* Update logback.version to v1.3.1 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/5906
* Update slf4j.version to v2.0.1 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/5907
* Update github/codeql-action digest to 904260d by @renovate in https://github.com/dropwizard/dropwizard/pull/5909
* Update dependency org.javassist:javassist to v3.29.2-GA by @renovate in https://github.com/dropwizard/dropwizard/pull/5910
* Update dependency org.liquibase:liquibase-core to v4.16.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/5911
* Update actions/stale digest to 99b6c70 by @renovate in https://github.com/dropwizard/dropwizard/pull/5908
* Update dependency org.apache.maven.plugins:maven-jar-plugin to v3.3.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5912
* Update dependency org.apache.maven.plugins:maven-jar-plugin to v3.3.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5913
* Update dependency org.apache.maven.plugins:maven-shade-plugin to v3.4.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5914
* Expose MethodInterceptor and the invoke method by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5880
* JavaDoc enhancements by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5749
* Expose `ManagedDataSource` to provide access to the connection pool by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5681
* Update dependency com.uber.nullaway:nullaway to v0.10.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/5934
* Update github/codeql-action digest to 86f3159 by @renovate in https://github.com/dropwizard/dropwizard/pull/5933
* Update dependency org.jdbi:jdbi3-bom to v3.33.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5979
* Update dependency org.testcontainers:testcontainers-bom to v1.17.4 by @renovate in https://github.com/dropwizard/dropwizard/pull/5973
* Update dependency org.hibernate:hibernate-core to v5.6.12.Final by @renovate in https://github.com/dropwizard/dropwizard/pull/5972
* Update actions/setup-java digest to a18c333 by @renovate in https://github.com/dropwizard/dropwizard/pull/5967
* Update github/codeql-action digest to e0e5ded by @renovate in https://github.com/dropwizard/dropwizard/pull/5968
* Update actions/cache action to v3.0.9 by @renovate in https://github.com/dropwizard/dropwizard/pull/5969
* Update dependency net.bytebuddy:byte-buddy to v1.12.17 by @renovate in https://github.com/dropwizard/dropwizard/pull/5970
* Update dependency org.apache.tomcat:tomcat-jdbc to v10.0.26 by @renovate in https://github.com/dropwizard/dropwizard/pull/5971
* Update actions/stale action to v6 by @renovate in https://github.com/dropwizard/dropwizard/pull/5980
* Skip Sonar execution on Windows runners by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5940
* Update junit5 monorepo to v5.9.1 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/5974
* Update slf4j.version to v2.0.3 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/5975
* Update dependency Sphinx to v5.2.3 by @renovate in https://github.com/dropwizard/dropwizard/pull/5976
* Update dependency org.apache.commons:commons-text to v1.10.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5977
* Update dependency org.apache.tomcat:tomcat-jdbc to v10.1.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/5978
* Remove the Jadira Usertype Core library by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/5939
* Update logback.version to v1.3.3 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/5999
* Update actions/cache action to v3.0.10 by @renovate in https://github.com/dropwizard/dropwizard/pull/5998
* Update dependency org.checkerframework:checker-qual to v3.26.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6005
* Update dependency org.jdbi:jdbi3-bom to v3.34.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6017
* Update dependency org.testcontainers:testcontainers-bom to v1.17.5 by @renovate in https://github.com/dropwizard/dropwizard/pull/6015
* Update actions/checkout action to v3.1.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6016
* Update actions/stale digest to 5ebf00e by @renovate in https://github.com/dropwizard/dropwizard/pull/6031
* Update github/codeql-action digest to 8075783 by @renovate in https://github.com/dropwizard/dropwizard/pull/6032
* Update logback.version to v1.3.4 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/6033
* Add manual workflow to trigger a release by @joschi in https://github.com/dropwizard/dropwizard/pull/6041
* Update actions/checkout digest to 93ea575 by @renovate in https://github.com/dropwizard/dropwizard/pull/6049
* Update dependency jp.skypencil.errorprone.slf4j:errorprone-slf4j to v0.1.16 by @renovate in https://github.com/dropwizard/dropwizard/pull/6050
* Update dependency net.bytebuddy:byte-buddy to v1.12.18 by @renovate in https://github.com/dropwizard/dropwizard/pull/6051
* Update dependency org.apache.tomcat:tomcat-jdbc to v10.1.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6052
* Update dependency org.cyclonedx:cyclonedx-maven-plugin to v2.7.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6053
* Update error_prone.version to v2.16 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/6055

## New Contributors
* @sleberknight made their first contribution in https://github.com/dropwizard/dropwizard/pull/5692
* @andref made their first contribution in https://github.com/dropwizard/dropwizard/pull/5506
* @sjthomason made their first contribution in https://github.com/dropwizard/dropwizard/pull/5748
* @bskarda made their first contribution in https://github.com/dropwizard/dropwizard/pull/5882

**Full Changelog**: https://github.com/dropwizard/dropwizard/compare/v3.0.0-beta.3...v3.0.0-beta.4

<hr /><em>This discussion was created from the release <a href='https://github.com/dropwizard/dropwizard/releases/tag/v3.0.0-beta.4'>v3.0.0-beta.4</a>.</em>

**Comment by brentryan** [comment]
What's the roadmap/plan look like for official release of 3.x?

**Reply by joschi** [reply]
@brentryan Unless any regressions are being reported, https://github.com/dropwizard/dropwizard/releases/tag/v3.0.0-beta.5 and https://github.com/dropwizard/dropwizard/releases/tag/v4.0.0-beta.4 will be released as Dropwizard 3.0.0 and 4.0.0 respectively.

**Reply by christopher-cudennec** [reply]
Do you also have a date when you want to publish v3 / v4? It almost sounds like you want to release them both at the same time.

**Reply by joschi** [reply]
> It almost sounds like you want to release them both at the same time.

Yes, that's the plan.

The main (almost only) difference is that Dropwizard 3.x still targets the Java EE namespace (`javax.*`) while Dropwizard 4.x targets the Jakarta EE namespace (`jakarta.*`).
